source/core/CEntryPoints.cpp

```diff
@@ -118,6 +118,45 @@ VIREO_EXPORT Int32 EggShell_PokeMemory(TypeManagerRef tm,
     }
 }
 //------------------------------------------------------------
+//! Allocates enough memory to fit a new object of TypeRef
+VIREO_EXPORT EggShellResult EggShell_AllocateData(TypeManagerRef tm, const TypeRef typeRef, void** dataRefLocation)
+{
+    TypeManagerScope scope(tm);
+    if (typeRef == nullptr || !typeRef->IsValid()) {
+        return kEggShellResult_InvalidTypeRef;
+    }
+
+    *dataRefLocation = nullptr;
+    Int32 topSize = typeRef->TopAQSize();
+    void* pData = THREAD_TADM()->Malloc(topSize);
```

Comment:
I remember Paul recommending to not use Malloc in Vireo in general. Is there a function in tm (TypeManagerRef) that will allocate memory for us given a TypeRef?

Reply:
Can we add a test to ensure that we are not leaking memory?

Reply:
Never mind, THREAD_TADM() is already using tm under the hood.